File: sp2mic-backend/sp2mic.WebAPI/Domain/Entities/Microsservico.cs
namespace sp2mic.WebAPI.Domain.Entities;

public class Microsservico
{
  public Microsservico()
  {
    //DtoClasses = new HashSet<DtoClasse>();
    Endpoints = new HashSet<Endpoint>();
  }

  public int Id {get; set;}
  public string NoMicrosservico {get; set;} = null!;
  public bool SnProntoParaGerar {get; set;}

  //public virtual ICollection<DtoClasse> DtoClasses {get; set;}
  public virtual ICollection<Endpoint> Endpoints {get; set;}
}

File: sp2mic-frontend/apps/sp2mic/src/app/shared/models/microsservico.ts
import { DtoClasse } from './dto-classe';
import { EndpointListagem } from './endpoint-listagem';

export class Microsservico {
  id: number;
  noMicrosservico: string;
  snProntoParaGerar: boolean;
  qtdEndpoints: number;
  dtoClasses?: DtoClasse[];
  endpoints?: EndpointListagem[];
}

export class MicrosservicoPaginado extends Microsservico {
  page: number;
  linesPerPage: number;
  orderBy: string;
  direction: string;
}
